File: static/CWE-798/java/patched.java
// Remediation for CWE-798: Use of Hard-coded Credentials
// Fix: Read database credentials from environment variables instead of embedding them.

import java.sql.Connection;
import java.sql.DriverManager;

class Patched_798 {
    public Connection connect(String database) throws Exception {
        String user = System.getenv("DB_USER");
        String password = System.getenv("DB_PASSWORD");
        if (user == null || password == null) {
            throw new IllegalStateException("DB credentials not configured");
        }
        String url = "jdbc:mysql://localhost/" + database;
        return DriverManager.getConnection(url, user, password);
    }
}

File: static/CWE-798/csharp/patched.cs
// Remediation for CWE-798: Use of Hard-coded Credentials
// Fix: Read the connection string from configuration and pass user input via a SQL parameter.

using System;
using System.Data.SqlClient;

namespace GuineaPig.Sast
{
    class Patched_798
    {
        public void Connect(string userInput)
        {
            string connectionString = Environment.GetEnvironmentVariable("APP_DB_CONNECTION");
            using (var conn = new SqlConnection(connectionString))
            {
                conn.Open();
                using (var cmd = new SqlCommand("SELECT 1 WHERE Tag = @tag", conn))
                {
                    cmd.Parameters.AddWithValue("@tag", userInput);
                    cmd.ExecuteNonQuery();
                }
            }
        }
    }
}
